File: 01-04-2026/wake_up_alarm.js
function alarmCheck(alarmTime, wakeTime) {
  const [atHr, atMin] = alarmTime.split(":").map(Number);
  const [wtHr, wtMin] = wakeTime.split(":").map(Number);

  if (atHr > wtHr || (atHr == wtHr && atMin > wtMin)) return "early";
  else if (alarmTime == wakeTime || (atHr == wtHr && wtMin - atMin <= 10))
    return "on time";
  else return "late";
}

console.log(alarmCheck("07:00", "06:45")); // "early".
console.log(alarmCheck("06:30", "06:30")); // "on time".
console.log(alarmCheck("08:10", "08:15")); // "on time".
console.log(alarmCheck("09:30", "09:45")); // "late".
console.log(alarmCheck("08:15", "08:25")); // "on time".
console.log(alarmCheck("05:45", "05:56")); // "late".
console.log(alarmCheck("04:30", "04:00")); // "early".

File: 01-04-2026/wake_up_alarm.py
def alarm_check(alarm_time, wake_time):
    atHr, atMin = map(int, alarm_time.split(":"))
    wtHr, wtMin = map(int, wake_time.split(":"))

    if (atHr > wtHr) or (atHr == wtHr and atMin > wtMin):
        return "early"
    elif alarm_time == wake_time or (atHr == wtHr and wtMin - atMin <= 10):
        return "on time"
    else: return "late"

print(alarm_check("07:00", "06:45")) # "early".
print(alarm_check("06:30", "06:30")) # "on time".
print(alarm_check("08:10", "08:15")) # "on time".
print(alarm_check("09:30", "09:45")) # "late".
print(alarm_check("08:15", "08:25")) # "on time".
print(alarm_check("05:45", "05:56")) # "late".
print(alarm_check("04:30", "04:00")) # "early".